### PR TITLE
[ci] Actually build all software in the sw_build job

### PR DESCRIPTION
**Pull request by a-will**
The sw_build job was missing many targets, as the build and test commands resolve wildcards very differently, even with --nobuild_tests_only.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -262,14 +267,11 @@ jobs:
...
-      for authenticity in "${AUTHENTICITY[@]}"; do
-        ROM=$(ci/scripts/target-location.sh "//sw/device/silicon_creator/rom:rom_with_${authenticity}_keys_fpga_cw310_scr_vmem")
-        cp $ROM "${ROM_TARGET}/$(basename "${ROM}")"
-        ROM_DIR=$(dirname $ROM)
-        cp "${ROM_DIR}/rom_with_${authenticity}_keys_fpga_cw310.elf" "${ROM_TARGET}/"
-        cp "${ROM_DIR}/rom_with_${authenticity}_keys_fpga_cw310.bin" "${ROM_TARGET}/"
-      done
+      # The deps 2 levels down from the opentitan_rom_binary include all the
+      # important files.
+      ROM_REAL_FILES=($(./bazelisk.sh outquery-all --noinclude_aspects --noimplicit_deps "deps(//sw/device/silicon_creator/rom:rom_with_real_keys_from_src_fpga_cw310, 2)" | sort | uniq))
```
Comment (a-will) [inline azure-pipelines.yml:272]: This took some experimentation... To make the correct query to get the ELF output file, we have to start from a node that is built for our usual target platform (the host) and follow the edges in the graph across the configuration transition (to riscv32).

**Review by rswarbrick — APPROVED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -267,14 +267,11 @@ jobs:
...
-      AUTHENTICITY=(fake real)
-      for authenticity in "${AUTHENTICITY[@]}"; do
-        ROM=$(ci/scripts/target-location.sh "//sw/device/silicon_creator/rom:rom_with_${authenticity}_keys_fpga_cw310_scr_vmem")
-        cp $ROM "${ROM_TARGET}/$(basename "${ROM}")"
-        ROM_DIR=$(dirname $ROM)
-        cp "${ROM_DIR}/rom_with_${authenticity}_keys_fpga_cw310.elf" "${ROM_TARGET}/"
-        cp "${ROM_DIR}/rom_with_${authenticity}_keys_fpga_cw310.bin" "${ROM_TARGET}/"
-      done
+      # The deps 2 levels down from the opentitan_rom_binary include all the
+      # important files.
```
Comment (marnovandermaas) [inline azure-pipelines.yml:271]: Nit: can we put this sentence on the same line?

File: azure-pipelines.yml (hunk 1 above)
Comment (marnovandermaas) [reply azure-pipelines.yml:272]: > To make the correct query to get the ELF output file, we have to start from a node that is built for our usual target platform (the host) and follow the edges in the graph across the configuration transition (to riscv32).

Maybe we can add this explanation as a comment in the code?

File: azure-pipelines.yml (hunk 2 above)
Comment (a-will) [reply azure-pipelines.yml:271]: Er... do we not attempt to do line limits in these files for readability? I know I didn't try very hard with some of the bash lines, but...

Comment (marnovandermaas) [reply azure-pipelines.yml:271]: I guess it's ok, just looks weird to me when there are other lines that are not wrapped. CI does not seem to mind either way, so I guess this need not be blocking.

File: azure-pipelines.yml (hunk 1 above)
Comment (a-will) [reply azure-pipelines.yml]: Done.

**Review by marnovandermaas — APPROVED** [review]
Thanks for this fix! Let's wait for CI to pass before merging.

**Comment by alphan** [comment]
Thanks @a-will !

**Pull request merged**